### ISSUE TITLE
Migration generates wrong down queries

### ISSUE THREAD

**Issue by tontonskie** [typeorm/typeorm#1001]
this is the up query
```
queryRunner.query("ALTER TABLE `kachick`.`users` CHANGE `currency` `currency` enum('usd', 'hkd') NOT NULL DEFAULT 'hkd'");
```

then this is the down which throw an error when I execute
```
queryRunner.query("ALTER TABLE `kachick`.`users` CHANGE `currency` `currency` enum(3)('usd', 'hkd') CHARACTER SET utf8 COLLATE utf8_general_ci NOT NULL DEFAULT usd");
```

I'm using MySQL database.
Up query is fine but when I execute down query it throws an error
```
executing query:  SELECT * FROM INFORMATION_SCHEMA.COLUMNS WHERE TABLE_SCHEMA = 'kachick' AND TABLE_NAME = 'migrations'
executing query:  SELECT * FROM `migrations` `migrations`
executing query:  START TRANSACTION
executing query:  ALTER TABLE `kachick`.`users` CHANGE `currency` `currency` enum(3)('usd', 'hkd') CHARACTER SET utf8 COLLATE utf8_general_ci NOT NULL DEFAULT usd
query failed:  ALTER TABLE `kachick`.`users` CHANGE `currency` `currency` enum(3)('usd', 'hkd') CHARACTER SET utf8 COLLATE utf8_general_ci NOT NULL DEFAULT usd
error:  { Error: ER_PARSE_ERROR: You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near '3)('usd', 'hkd') CHARACTER SET utf8 COLLATE utf8_general_ci NOT NULL DEFAULT usd' at line 1
    at Query.Sequence._packetToError (/Users/tonton/projects/web/kachick-server/node_modules/mysql/lib/protocol/sequences/Sequence.js:52:14)
    at Query.ErrorPacket (/Users/tonton/projects/web/kachick-server/node_modules/mysql/lib/protocol/sequences/Query.js:77:18)
    at Protocol._parsePacket (/Users/tonton/projects/web/kachick-server/node_modules/mysql/lib/protocol/Protocol.js:279:23)
    at Parser.write (/Users/tonton/projects/web/kachick-server/node_modules/mysql/lib/protocol/Parser.js:76:12)
    at Protocol.write (/Users/tonton/projects/web/kachick-server/node_modules/mysql/lib/protocol/Protocol.js:39:16)
    at Socket.<anonymous> (/Users/tonton/projects/web/kachick-server/node_modules/mysql/lib/Connection.js:103:28)
    at emitOne (events.js:115:13)
    at Socket.emit (events.js:210:7)
    at addChunk (_stream_readable.js:266:12)
    at readableAddChunk (_stream_readable.js:253:11)
    --------------------
    at Protocol._enqueue (/Users/tonton/projects/web/kachick-server/node_modules/mysql/lib/protocol/Protocol.js:145:48)
    at PoolConnection.query (/Users/tonton/projects/web/kachick-server/node_modules/mysql/lib/Connection.js:208:25)
    at MysqlQueryRunner.<anonymous> (/Users/tonton/projects/web/kachick-server/src/driver/mysql/MysqlQueryRunner.ts:191:36)
    at step (/Users/tonton/projects/web/kachick-server/node_modules/typeorm/driver/mysql/MysqlQueryRunner.js:32:23)
    at Object.next (/Users/tonton/projects/web/kachick-server/node_modules/typeorm/driver/mysql/MysqlQueryRunner.js:13:53)
    at fulfilled (/Users/tonton/projects/web/kachick-server/node_modules/typeorm/driver/mysql/MysqlQueryRunner.js:4:58)
    at <anonymous>
    at process._tickDomainCallback (internal/process/next_tick.js:228:7)
  code: 'ER_PARSE_ERROR',
  errno: 1064,
  sqlMessage: 'You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near \'3)(\'usd\', \'hkd\') CHARACTER SET utf8 COLLATE utf8_general_ci NOT NULL DEFAULT usd\' at line 1',
  sqlState: '42000',
  index: 0,
  sql: 'ALTER TABLE `kachick`.`users` CHANGE `currency` `currency` enum(3)(\'usd\', \'hkd\') CHARACTER SET utf8 COLLATE utf8_general_ci NOT NULL DEFAULT usd' }
executing query:  ROLLBACK
```

**Comment by luchillo17** [comment]
Maybe it's the `(3)` part, try and remove it manually, if so then we know where to start looking.

**Comment by pleerock** [comment]
Looks like a bug. We should not insert `length` for `enum` column type.

**Comment by tontonskie** [comment]
@pleerock I don't know why the down query has `CHARACTER SET utf8 COLLATE utf8_general_ci` and the default value is not quoted `DEFAULT usd` 

when I generate the migration I have `charset: 'utf8_unicode_ci'` in connection options

**Comment by AlexMesser** [comment]
Rignt now I'm working on migration processes and queries. There is many changes, refactoring and fixes. I plan to finish it at the end of next week.

**Comment by pleerock** [comment]
I think this will be postponded for 0.2.0, and @AlexMesser is still working on it afaik

**Comment by AlexMesser** [comment]
fixed and released in `typeorm@0.2.0-alpha.23`. You can try it via `npm i typeorm@next`.
